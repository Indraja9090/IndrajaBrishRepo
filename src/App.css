/* ----- Built-in CSS code that comes along with CRA command-----*/ 
/* .App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
} */

/* Code related to article https://keploy.io/blog/community/a-guide-to-testing-react-components-with-jest-and-react-testing-library#let-s-start-writing-testcases -----*/
.app-container {
  width: 400px;
  margin: 50px auto;
  font-family: Arial, sans-serif;
  background: #f7f7f7;
  padding: 20px;
  border-radius: 8px;
}

.aadhar-card {
  max-width: 400px;
  margin: 20px auto;
  padding: 20px 25px;
  border-radius: 12px;
  background-color: #f9f9f9;
  box-shadow: 0 4px 10px rgba(0, 0, 0, 0.1);
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
  color: #333;
}

.aadhar-card h2 {
  text-align: center;
  margin-bottom: 20px;
  color: #1a237e;
  font-size: 22px;
}

.aadhar-card p {
  margin: 10px 0;
  font-size: 16px;
}

.aadhar-card strong {
  color: #555;
}


