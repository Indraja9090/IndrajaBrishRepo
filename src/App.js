//App.js
import './App.css';
import AadharProfile from './components/aadharProfile';
function App() {
  return (
    <div className="app-container">
      <h1>Snapshot testing on components</h1>
      <AadharProfile
        name="Indraja"
        aadharNumber="1234-5678-9012"
        dob="01-01-1990"
        gender="Female"
      />
    </div>
  );
}

export default App;