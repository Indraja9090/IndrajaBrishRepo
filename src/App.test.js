//App.test.js
import { render, screen } from '@testing-library/react';
import App from './App';

test('renders heading for snapshot testing', () => {
  render(<App />);
  const linkElement = screen.getByText(/Snapshot testing on components/i);
  expect(linkElement).toBeInTheDocument();
});

/* snapshot testing*/
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';

test('matches snapshot', () => {
  /* You’re telling React Testing Library to give you a "snapshot fragment" — basically, a snapshot of the entire rendered DOM tree at that point in time.*/
  // const { asFragment } = render(<App />);

  // expect(asFragment()).toMatchSnapshot();

  const tree = renderer.create(<MyComponent />).toJSON();
  expect(tree).toMatchSnapshot();

});



