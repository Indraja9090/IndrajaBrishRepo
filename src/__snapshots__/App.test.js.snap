// Jest Snapshot v1, https://goo.gl/fbAQLP
//App.test.js.snap
exports[`App Snapshot renders App component with AadharProfile correctly 1`] = `null`;
